common/metrics/events_userscope.go

```diff
@@ -66,27 +74,38 @@ func (e eventsUserScope) StartTimer(timer string) Stopwatch {
 }
 
 // RecordTimer records a timer for the given metric name
-func (e eventsUserScope) RecordTimer(timer string, d time.Duration) {
-	e.provider.Timer(timer, nil).Record(d)
+func (e *eventsUserScope) RecordTimer(timer string, d time.Duration) {
+	e.provider.Timer(timer, &MetricOptions{
+		Unit: event.UnitMilliseconds,
+	}).Record(d)
 }
 
 // RecordDistribution records a distribution (wrapper on top of timer) for the given
 // metric name
-func (e eventsUserScope) RecordDistribution(id string, unit MetricUnit, d int) {
+func (e *eventsUserScope) RecordDistribution(id string, unit MetricUnit, d int) {
 	e.provider.Histogram(id, &MetricOptions{
 		Unit: event.Unit(unit),
 	}).Record(int64(d))
 }
 
 // UpdateGauge reports Gauge type absolute value metric
-func (e eventsUserScope) UpdateGauge(gauge string, value float64) {
+func (e *eventsUserScope) UpdateGauge(gauge string, value float64) {
 	e.provider.Gauge(gauge, nil).Record(value)
 }
 
 // Tagged returns a new scope with added and/or overriden tags values that can be used
 // to provide additional information to metrics
-func (e eventsUserScope) Tagged(tags map[string]string) UserScope {
-	return newEventsUserScope(e.provider.WithTags(mapToTags(tags)...))
+func (e *eventsUserScope) Tagged(tags map[string]string) UserScope {
+	switch {
+	case tags != nil && e.tags == nil:
+		return newEventsUserScope(e.provider, tags)
+	case tags == nil && e.tags != nil:
+		return newEventsUserScope(e.provider, e.tags)
+	default:
+		m := maps.Clone(e.tags)
+		maps.Copy(m, tags)
+		return newEventsUserScope(e.provider, m)
+	}
```

Comment:
would it be better to check len(map) == 0?
And do we care the case both are empty?